electrum/plugins/trustedcoin/qml.py

```diff
@@ -0,0 +1,332 @@
+                try:
+                    self.plugin.logger.debug(f'check OTP, shortId={short_id}, otp={otp}')
+                    server.auth(short_id, otp)
+                except TrustedCoinException as e:
+                    if e.status_code == 400:  # invalid OTP
+                        self.plugin.logger.debug('Invalid one-time password.')
+                        self.otpError.emit()
+                    else:
+                        self.plugin.logger.error(str(e))
+                        self._createRemoteKeyError = str(e)
+                        self.createRemoteKeyErrorChanged.emit()
+                except Exception as e:
+                    self.plugin.logger.error(str(e))
+                    self._createRemoteKeyError = str(e)
+                    self.createRemoteKeyErrorChanged.emit()
+                else:
+                    self.plugin.logger.debug('OTP verify success')
+                    self.otpSuccess.emit()
+                finally:
+                    self._busy = False
+                    self.busyChanged.emit()
+
+            self._busy = True
+            self.busyChanged.emit()
+            t = threading.Thread(target=check_otp_task)
+            t.daemon = True
+            t.start()
+
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    @hook
+    def load_wallet(self, wallet: 'Abstract_Wallet'):
+        if not isinstance(wallet, self.wallet_class):
+            return
+        self.logger.debug(f'plugin enabled for wallet "{str(wallet)}"')
+        #wallet.handler_2fa = HandlerTwoFactor(self, window)
+        if wallet.can_sign_without_server():
+            self.so._canSignWithoutServer = True
+            self.so.canSignWithoutServerChanged.emit()
+
+            msg = ' '.join([
+                _('This wallet was restored from seed, and it contains two master private keys.'),
+                _('Therefore, two-factor authentication is disabled.')
+            ])
+            #action = lambda: window.show_message(msg)
+        #else:
+            #action = partial(self.settings_dialog, window)
+        #button = StatusBarButton(read_QIcon("trustedcoin-status.png"),
+                                 #_("TrustedCoin"), action)
+        #window.statusBar().addPermanentWidget(button)
+        self.start_request_thread(wallet)
+
+    @hook
+    def init_qml(self, gui: 'ElectrumGui'):
+        self.logger.debug(f'init_qml hook called, gui={str(type(gui))}')
+        self._app = gui.app
+        # important: QSignalObject needs to be parented, as keeping a ref
+        # in the plugin is not enough to avoid gc
+        self.so = Plugin.QSignalObject(self, self._app)
+
+        # extend wizard
+        self.extend_wizard()
+
+    def extend_wizard(self):
+        wizard = self._app.daemon.newWalletWizard
+        self.logger.debug(repr(wizard))
+        views = {
+            'trustedcoin_start': {
+                'gui': '../../../../plugins/trustedcoin/qml/Disclaimer',
```

Comment:
what is this relative to -- what is the "current directory" here?

Reply:
Current directory is the directory where the QML file exists that is referencing it. 

SO
```
/path/1.qml: Qt.resolvedUrl('path/2.qml')
/path/path/2.qml: Qt.resolvedUrl('../1.qml')
```

I know, this is ugly as f :( I hope Qt6 improves on that.
I've tried to add more import paths to the QML engine, to have more 'roots', but that doesn't work :(

Reply:
But on the bright side, if we factor out plugins into the main tree, this disappears mostly.

Reply:
Note: I also tried to define a [QQmlAbstractUrlInterceptor](https://doc.qt.io/qt-5/qqmlabstracturlinterceptor.html#details), but that didn't really work in python/PyQt.

Reply:
Relative paths can also be resolved by packaging all `qml` files in a resource bundle. 
This however requires one to regenerate the bundle before testing each change.